app/src/main/java/org/mozilla/fenix/settings/deletebrowsingdata/DeleteBrowsingDataFragment.kt

```diff
@@ -239,6 +239,34 @@ class DeleteBrowsingDataFragment : Fragment(R.layout.fragment_delete_browsing_da
             fragmentView.site_permissions_item
         )
     }
+    // If coming from Open Tab -> Settings, pop back to Home
+    // If coming from Home -> Settings, pop back to Settings
+    private fun returnToDeletionOrigin() {
+        if (checkIfBrowserFragmentInBackStack()) findNavController().popBackStack(R.id.homeFragment, false)
+        else findNavController().popBackStack()
+    }
+
+    // For some reason, the only way you can tell if you came from
+    // BrowserFrag -> Settings OR BrowserFrag -> HomeFrag -> Settings is to check whether
+    // there is a browser fragment entry in the back stack. If there is, it means you came from BrowserFrag -> Settings
+    // if there isn't, it means you came from HomeFrag -> Settings
+    private fun checkIfBrowserFragmentInBackStack(): Boolean {
+        val backStackEC = parentFragmentManager.backStackEntryCount
+        val index = backStackEC - 1
```

Comment:
For future reference, you can deal with these type of off-by-one errors by using `0 until index` instead of `0..index`. Also, `index` isn't a great name. `i` is serving as the index here. `index` is simply the back stack entry count's maximum index.